staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -762,12 +762,27 @@ func shouldDelegateList(opts storage.ListOptions) bool {
 	consistentReadFromStorage := resourceVersion == "" && !(consistentListFromCacheEnabled && requestWatchProgressSupported)
 	// Watch cache doesn't support continuations, so serve them from etcd.
 	hasContinuation := len(pred.Continue) > 0
-	// Serve paginated requests about revision "0" from watch cache to avoid overwhelming etcd.
-	hasLimit := pred.Limit > 0 && resourceVersion != "0"
```

Comment:
I think that we don't have to check if `storage.RequestWatchProgress)` is supported (NotOlderThan, Any semantics).

Reply:
agree